packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1459,7 +1460,7 @@ abstract class DiagnosticsNode {
   ///
   /// If `minLevel` is [DiagnosticLevel.hidden] no diagnostics will be filtered.
   /// If `minLevel` is [DiagnosticLevel.off] all diagnostics will be filtered.
-  bool isFiltered(DiagnosticLevel minLevel) => level.index < minLevel.index;
+  bool isFiltered(DiagnosticLevel minLevel) => kReleaseMode ? true : level.index < minLevel.index;
```

Comment:
This filters everything out of release mode but I suspect not in a way that helps tree shaking. Perhaps I'm missing the intent of this change?

Reply:
When I re-compiled with this set, I saw fewer bytes in the snapshots.  It's possible that I didn't do this in isolation though.

My thought is that making these properties constant in release mode is helping tree shaking.

Reply:
In my experiments about half a year ago the tree shaker was not smart enough to act on the always-true or always-false return value of a method. Did anything change there?

Reply:
Apparently something did.  It's a small difference but it does take some size off.